assignment-2/src/view/SmartQuestionsList.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import userModel from "../model/userModel";
+import QuestionsList from "./QuestionsList"
+import questionModel from "../model/questionModel"
+import questionsListPresenter from "../presenter/questionsListPresenter";
+
+const mapModelStateToComponentState = modelState => ({
+    questions : modelState.questions,
+    title: modelState.newQuestion.title,
+    text: modelState.newQuestion.text,
+    tags: modelState.newQuestion.tags
+});
+
+const comparator = function(question1, question2)  {
+    if (question1.creationDate > question2.creationDate){
+        return 1;
+    }
+    if (question1.creationDate < question2.creationDate){
+        return -1;
+    }
+    return 0;
+}
+
+export default class SmartQuestionsList extends Component {
+    constructor(){
+        super();
+        this.state = mapModelStateToComponentState(questionModel.state);
+        this.state.questions.sort(comparator);
```

Comment:
This changes the array directly (which also affects the model, as you are just using the same array). You create a copy and then sort it (inside the `mapModelStateToComponentState` method).